File: CF PABC.NET/G.pas
begin
  var M := ReadlnInteger();
  var LayersNum := (2 ** M).Trunc();
  var F := ReadArrInteger(LayersNum);
  
  Println(2);
  Println(LayersNum, 1);
  
  for var I := 0 to LayersNum - 1 do
  begin
    var Bin := System.Convert.ToString(I, 2).PadLeft(M, '0').Select(C -> C.ToDigit()).ToList();
    var BinSum := 0.5 - Bin.Sum();
    Bin := Bin.Select(Digit -> Digit = 1 ? 1 : -1).ToList();
    Bin.Print();
    Write(' ');
    BinSum.Println();
  end;
  
  F.Print();
  Write(' ');
  (-0.5).Print();
end.

File: CF Python/G.py
m = int(input())
layers_num = 2 ** m
f = [int(input()) for i in range(layers_num)]

print(2)
print(layers_num, 1)
for i in range(layers_num):
    binary = [1 if c == '1' else 0 for c in f'{i:0{m}b}']
    binary_sum = 0.5 - sum(binary)
    binary = [1 if digit == 1 else -1 for digit in binary]
    print(*binary, binary_sum)

print(*f, -0.5)
